packages/amplify-schema-validator/src/validators/use-directives-from-newer-transformer-version.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  DocumentNode,
+  Kind,
+  ObjectTypeDefinitionNode,
+} from 'graphql';
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+
+/**
+ * Validates that @belongsTo, @connection directives from older graphql transformer version is not used in version 1.0
+ *
+ * @param schema graphql schema
+ * @returns true if correct directives are used
+ */
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+const GRAPHQL_TRANSFORMER_V2_DIRECTIVES = ['hasOne', 'index', 'primaryKey', 'belongsTo', 'manyToMany', 'hasMany', 'default'];
+export const validateDirectivesFromNewerTransformerVersionAreNotUsed = (
+  schema: DocumentNode, amplifyFeatureFlags?: string, _dataStoreEnabled?: boolean,
+): Error[] => {
+  if (!amplifyFeatureFlags) {
+    return [];
+  }
+  const featureFlags = JSON.parse(amplifyFeatureFlags);
```

Comment:
As mentioned in the overall comment, this lib shouldn't be responsible for the parse + check to determine transformer version, but the logic here is also not quite right. The actually behavior is to check `useExperimentalPipelinedTransformer && transformerVersion === 2`.

E.g. https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-category-api/src/graphql-transformer/transformer-version.ts#L15-L37

Reply:
Checking `useExperimentalPipelinedTransformer && transformerVersion === 2` behavior in CP and passing the resolved graphql version